packages/language-server/src/completion-items.ts

```diff
@@ -60,10 +75,11 @@ function transformToLspSuggestions(
     if (suggestion.ui5Node.deprecatedInfo?.isDeprecated) {
       detailText = `(deprecated) ${detailText}`;
     }
+
     const completionItem: CompletionItem = {
       label: suggestion.ui5Node.name,
-      // TODO use textEdit instead of insertText to support replacing "sap.m.Button" with "Button"
-      insertText: createInsertText(suggestion),
+      filterText: getNodeFilterText(suggestion),
```

Comment:
What would happen if filterText === textEdit always? what won't work?

Reply:
The filterText should contain the prefix (replaced text until the cursor position) and is used for sorting too (for example results that have the prefix as a substring rank higher than results that have additional characters in the middle)

Reply:
> The filterText should contain the prefix (replaced text until the cursor position) 
 
- Does it contain anything else?
- Could we compute it more simply using offsets and substring on `insertText` in that case?

Reply:
Moved the logic to `createTextEdit` according to what we discussed (it didn't add as much code as I thought so I didn't refactor it)